app/Providers/AuthServiceProvider.php

```diff
@@ -29,7 +29,17 @@ public function boot()
 
         Passport::routes();
 
-        Passport::tokensExpireIn(Carbon::now()->addMinute(10));
+        Route::post('oauth/token', [
+            'middleware' => 'password-grant',
+            'uses' => '\Laravel\Passport\Http\Controllers\AccessTokenController@issueToken'
+        ]);
+        Route::post('oauth/token/refresh', [
+            'middleware' => ['web', 'auth', 'password-grant'],
+            'uses' => '\Laravel\Passport\Http\Controllers\TransientTokenController@refresh'
+        ]);
+
+        $tokenLifeTime = env('APP_ENV') === 'local' ? Carbon::now()->addYear() : Carbon::now()->addMinute(10);
+        Passport::tokensExpireIn($tokenLifeTime);
```

Comment:
Can be simplified to : 

```php
Passport::tokensExpireIn(Carbon::now()->addMinute(10));
``` 

?

Reply:
How so ?

Reply:
Set the token lifetime to 10 minutes for every env by removing line 41 and changing Passport::tokenExpireIn to : 

```php
Passport::tokensExpireIn(Carbon::now()->addMinute(10));
```

Reply:
Yeah no working with Postman it's really a pain in the ass the generate a token every 10 minutes manually. Plus it's just an env variable to modify. The logic of refreshing token behind is not altered.